File: 201-300/242-valid-anagram.cpp
bool isAnagram(string s, string t) {
    vector<int> count(26, 0);
    if (s.size() != t.size()) {
        return false;
    }
    for (int i = 0; i < s.size(); ++i) {
        ++count[s[i] - 'a'];
        --count[t[i] - 'a'];
    }

    for (const auto &n : count) {
        if (n != 0) return false;
    }

    return true;
}

File: python version/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        word_dict = {}
        for char in s:
            if char not in word_dict:
                word_dict[char] = 1
            else:
                word_dict[char] += 1
        
        for char in t:
            if char not in word_dict:
                return False
            word_dict[char] -= 1
            if word_dict[char] < 0:
                return False

        return True